app/views/certificates/pathways/_recommended_courses.html.erb

```diff
@@ -25,10 +25,12 @@
               <p class="ncce-pathway-courses__location <%= activity_icon_class(pa) %> govuk-!-margin-top-0">
                 <%= activity_type(pa) %>
               </p>
-              <p class="ncce-pathway-courses__location icon-clock govuk-!-margin-top-0">
-                <% course = Achiever::Course::Template.find_by_activity_code(pa.stem_activity_code)%>
-                <%= pluralize(course.duration_value, course.duration.downcase.chomp('s')) if course.duration.present? %> 
-              </p>
+              <% course = Achiever::Course::Template.maybe_find_by_activity_code(pa.stem_activity_code)%>
+              <% if (duration = course&.duration.presence) && (duration_value = course&.duration_value.presence) %>
+                <p class="ncce-pathway-courses__location icon-clock govuk-!-margin-top-0">
+                  <%= pluralize(duration_value, duration.downcase.chomp('s')) %>
+                </p>
+              <% end %>
```

Comment:
Could we add a method in the template class for `duration_present?` and `formatted_duration` so this code needs less knowledge of a course's internals?

```suggestion
              <% if course.duration_present? %>
                <p class="ncce-pathway-courses__location icon-clock govuk-!-margin-top-0">
                  <%= course.formatted_duration %>
                </p>
              <% end %>
```

Reply:
This is not really in scope so I'm happy for it to be moved to a ticket for a later date

Reply:
Yes - out-of-scope - (not least because I don't have a fixed IP today to test against Dynamics.)